Jenkinsfile

```diff
@@ -55,14 +55,27 @@ node {
 onlyOnMaster {
     milestone()
     stage('qa deploy') {
-        deployApp(image: img, app: 'h', env: 'qa')
+        deployApp(image: img, app: 'h', env: 'qa', region: "us-west-1")
     }
 
     milestone()
-    stage('prod deploy') {
-        input(message: "Deploy to prod?")
-        milestone()
-        deployApp(image: img, app: 'h', env: 'prod')
+    stage("approval") {
+        input(message: "Proceed to production deploy?")
+    }
+
+    milestone()
+    stage("prod Deploy") {
+        lock("prod deploy") {
+            parallel(
+                us: {
+                    sleep 2
+                    deployApp(image: img, app: "h", env: "prod", region: "us-west-1")
+                },
+                ca: {
+                    deployApp(image: img, app: "h-ca", env: "prod", region: "ca-central-1")
+                }
+            )
+        }
     }
```

Comment:
A stage dedicated to production deploys:

1. A `parallel` block has been introduced. When called, it will make two calls to `deployApp`. The first deals with deploying `us-west-1` and the second deals with deploying `ca-central-1`.
2. The `sleep 2` command in the `us` block is required to insert a small pause in the calls to `deployApp`. Without a pause it can end up running the first job twice.

Reply:
> The sleep 2 command in the us block is required to insert a small pause in the calls to deployApp. Without a pause it can end up running the first job twice.

Can you expand on this? Do you know why this can happen and whether `sleep 2` is guaranteed to work?

Reply:
I discovered the problem when setting up the new Via environments that deploy to two Elastic Beanstalk environments in a similar manner.

I don't know why this happens, but what I can say is the Via deploys have not been failing since this was deployed a month to 6 weeks ago.

I attempted to recreate the problem on a local Jenkins instance I run, but unfortunately I wasn't able to. I setup the following during my testing

1. Created a deploy pipeline. It doesn't do much other than `echo` information it was called with. 
2. Created a build pipeline with a couple of stages. As part of one of these stages it calls the deploy pipeline twice in a parallel block. The following code was used...

```
parallel(
    one: {
        build job: 'deploy', parameters: [string(name: 'APP', value: 'one'), string(name: 'REGION', value: 'ca-central-1')]
    },
    two: {
        build job: 'deploy', parameters: [string(name: 'APP', value: 'two'), string(name: 'REGION', value: 'us-west-1')]
    }
)
```
  
The above works without issue, and is similar code to the [deployApp.groovy](https://github.com/hypothesis/pipeline-library/blob/master/vars/deployApp.groovy) shared library. The only difference I can think of is the way the code gets imported.

I have a suspicion that we could reduce the pause from 2 seconds to practically nothing, but that remains untested. During my original testing I remember setting it to 10 seconds that worked so I reduced it to 2.

Let me know what you think? I can dig in further and create a local workflow that uses shared libraries in an attempt to diagnose the issue. My only concern is the time it may take. Maybe this is something we should create an issue for, document and revisit once we have Canada completed.